pyqt-apps/siriushla/li_ap_mps/main.py

```diff
@@ -0,0 +1,533 @@
+        return lay
+
+    def genStringTempPV(self, name, prefix_num, name_num):
+        ''' Generate Temperature PV name '''
+        return name + str(prefix_num) + 'Temp' + str(name_num)
+
+    def getPVConfig(self, config, index):
+        ''' Get if the PV has configuration 1 or 0'''
+
+        if type(config) == list:
+            return config[index]
+        else:
+            return config
+
+    def getPVControl(self, is_control, index):
+        ''' Get if the PV is a control or a status'''
+
+        if type(is_control) == list:
+            return is_control[index]
+        else:
+            return is_control
+
+    def countGen(self, count, val):
+        ''' Counter for water data positioning '''
+        if count[1] >= val:
+            count[0] += 1
+            count[1] = 1
+        else:
+            count[1] += 1
+        return count
+
+    def countVA(self, count):
+        ''' Counter for vaccum data positioning '''
+
+        if count in [
+            [2, 3], [3, 3], [6, 3],
+            [11, 3], [12, 3]
+        ]:
+            count[1] += 1
+        if count in [
+            [4, 1], [5, 1], [8, 1],
+            [10, 1], [13, 1], [15, 1]
+        ]:
+            count[1] += 3
+
+        count = self.countGen(count, 4)
+        return count
+
+    def countWater(self, count):
+        if count in [
+            [2, 5], [3, 5]
+        ]:
+            count[1] += 3
+        count = self.countGen(count, 6)
+        return count
+
+    def countTemp(self, count, title):
+        ''' Counter for temperature data positioning '''
+        if title == 'Water Temperature':
+            orient = [1, 0, 6]
+        else:
+            orient = [0, 1, 2]
+
+        if count[orient[1]] >= orient[2]:
+            count[orient[0]] += 1
+            count[orient[1]] = 1
+        else:
+            count[orient[1]] += 1
+        return count
+
+    def updateCount(self, count, title):
+        ''' Update counter for data positioning '''
+        if title == 'Klystrons':
+            count = self.countGen(count, 4)
+        elif title == 'Water':
+            count = self.countWater(count)
+        elif title == 'Modulator':
+            count = self.countGen(count, 2)
+        elif title == 'Gate Valve':
+            count = self.countGen(count, 5)
+        elif title in ['Temperature', 'Water Temperature']:
+            self.countTemp(count, title)
+        elif title == 'VA':
+            count = self.countVA(count)
+        else:
+            count = self.countGen(count, 1)
+
+        return count
+
+    def setTitleLabel(self, item, axis, layout):
+        ''' Display title labels '''
+        pos = [0, 0]
+        for pos[axis] in range(1, len(item)+1):
+            lbl_header = QLabel('<h4>'+item[pos[axis]-1]+'</h4>')
+            lbl_header.setAlignment(Qt.AlignCenter)
+            layout.addWidget(lbl_header, pos[0], pos[1], 1, 1)
+            if axis == 0:
+                lbl_header.setMaximumWidth(50)
```

Comment:
I think it is better to use here the SizePolicy and layout alignment, this way we do not fix sizes in a absolut manner, what can be a problem with different display sizes.
I prefer:
```            
for pos[axis] in range(1, len(item)+1):
    lbl_header = QLabel('<h4>'+item[pos[axis]-1]+'</h4>')
    layout.addWidget(lbl_header, pos[0], pos[1], 1, 1, alignment=Qt.AlignCenter)
    if axis == 0:   
        lbl_header.setSizePolicy(QSizePolicy.Maximum, QSizePolicy.Maximum)